### DISCUSSION TITLE
Creating a wallet for a user before they log in.

### DISCUSSION THREAD

**Discussion by zer0stars** [Web3Auth/Web3Auth#1216] [Help]
**Issue title**: Pre-create a wallet and retrieve public key.
**Describe the question**
One of our use cases for an in-app wallet is to simplify onboarding via a share link. When a user wants to share with another user who has not created a wallet yet, we want to be able to pre-create the wallet for newuser2@gmail.com and get the associated 0x address.

**Comment by r-muresan** [comment]
@zer0stars Have you discovered a solution for this?

**Comment by gofman8** [comment]
Also interested

**Comment by BodySplash** [comment]
Very interested by that one too. 
It was a nice feature of plain old [vanilla torus](https://docs.tor.us/wallet/api-reference/address-resolver). I understand why now keys are scoped by dApp (clientId & verifiers), but it would be incredibly useful to be able to do the same with our own clientId.
